File: Daily/Rust/Leetcode1588-所有奇数长度子数组的和.rs
impl Solution {
    pub fn sum_odd_length_subarrays(mut arr: Vec<i32>) -> i32 {
        let mut a = vec![0;1];
        a.append(&mut arr);
        let n = a.len();
        for i in 1..n{
            a[i] += a[i - 1];
        }
        let mut res = 0;
        for i in 1..n{
            for j in i..n{
                if (j - i + 1) & 1 == 1{
                    res += a[j] - a[i - 1];
                }
            }
        }
        res
    }
}

File: Leetcode/sbgo/Leetcode1588-所有奇数长度子数组的和.go
package sbgo

func sumOddLengthSubarrays(arr []int) (res int) {
	arr = append([]int{0}, arr...)
	for i := 1; i < len(arr); i++ {
		arr[i] += arr[i-1]
	}
	for i := 1; i < len(arr); i++ {
		for j := i; j < len(arr); j++ {
			if (j-i+1)&1 == 1 {
				res += arr[j] - arr[i-1]
			}
		}
	}
	return
}
